internal/pkg/agent/vault/aesgcm/aesgcm.go

```diff
@@ -62,7 +62,7 @@ func Encrypt(key, data []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	aesGCM, err := cipher.NewGCM(block)
+	aesGCM, err := cipher.NewGCMWithRandomNonce(block)
```

Comment:
This is the very early stages of trying to use the new method ([documented here](https://pkg.go.dev/crypto/cipher@master#NewGCMWithRandomNonce)).

We will probably need to do a bit more here as the returned aesGCM behaves differently then the current one.